hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/ws/server/FeedCommWebSocket.java

```diff
@@ -66,19 +63,17 @@
     @OnOpen
     public void feedSessionOpen(Session session, @PathParam("feedId") String feedId) {
         MsgLogger.LOG.infoFeedSessionOpened(feedId, session.getId());
-        boolean successfullyAddedSession = connectedFeeds.addSession(feedId, session);
```

Comment:
If we change the way the UI clients connect, we could do the same thing (and thus use the same implementation) as the feeds. For example, today UI clients connect to the WS via "ws://<host>:8080/hawkular/command-gateway/ui/ws - its the same URL for all UI clients. For feeds, they connect to ".../feed/<feedID>" We could do the same for UI clients - so they connect to something like "ws://<host>:8080/hawkular-command-gateway/ui/<ui-client-id>

We could even use that ui-client-id as the identifier of the client so you don't need that "sender ID" in the JSON. We already know "this session is associatd with this UI client".

The only issue there is the UI would be required to generate some unique identifier, but I'm sure javascript can generate some kind of UUID and use that.